### PR TITLE
Refactor to introduce Worker, Users, Tasks and YAML config

### PR DESCRIPTION
**Pull request by lmars**
This is a major refactor that introduces a few new concepts:

- a Worker that performs the same function as a [Python Locust worker](https://docs.locust.io/en/stable/running-locust-distributed.html)
- a User that performs subscribe, publish or presence tasks
- a YAML config file
- channel config that use Go templates
- Redis to optionally assign unique numbers to users across multiple workers
- a set of docker-compose based examples

It's probably easiest to point people at the [updated README](https://github.com/ably/ably-boomer/blob/refactor/README.md) which explains the new usage in more details.

I appreciate that this is a lot, but I just went full throttle :joy:

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,19 +4,28 @@ go 1.14
 
 require (
 	github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d // indirect
-	github.com/ably-forks/boomer v0.0.0-20200717132144-47e73943070d
-	github.com/ably/ably-go v1.1.6-0.20200905151113-60928e378b37
+	github.com/ably-forks/boomer v0.0.0-20210109192149-bcdec6481b67
```
Comment (JamesFrost) [inline go.mod:7]: Looks like this points to a branch - perhaps we should consider merging into master?

https://github.com/myzhan/boomer/compare/master...ably-forks:local-spawn-event

**Review by JamesFrost — APPROVED** [review]
A much needed refactor with lots of improvements - nice work!

Will be interested to see it run a large scale test 😎

File: go.mod (hunk 1 above)
Comment (lmars) [reply go.mod:7]: Yeh I opened an upstream PR https://github.com/myzhan/boomer/pull/113, hoping that will get merged and we can drop the fork.

**Comment by paddybyers** [comment]
> Will be interested to see it run a large scale test 😎

You're volunteering?

File: go.mod (hunk 1 above)
Comment (JamesFrost) [reply go.mod:7]: Ah nice, I didn't see that. Dropping the fork will be good 👍

**Comment by JamesFrost** [comment]
@paddybyers I'm a glutton for punishment 😉

**Review by Jmgr — APPROVED** [review]
Impressive work, well done :+1:

**Comment by lmars** [comment]
✊

**Pull request merged**